### DISCUSSION TITLE
Home Assistant - PVE CPU Type

### DISCUSSION THREAD

**Discussion by JPurp** [tteck/Proxmox#1546] [Q&A] [answered]
With the update to PVE 8 the default CPU type changed to "x86-64-v2-AES" from the previous "kvm64" default.

Any thoughts/concerns about changing the CPU type to the new default? 

Could the script be updated to offer this option?

Thanks
-JP

**Comment by tteck** [comment, marked as answer]
I had a similar thought, but it's important to note that `x86-64-v2-AES` is an alternative option and not the default. If you examine the code snippet below, you'll see that the script sets the default `CPU_TYPE` as an empty string (`CPU_TYPE=""`), which utilizes Proxmox's default value, which is still `kvm64`.

```
  if CPU_TYPE1=$(whiptail --title "CPU MODEL" --radiolist "Choose" --cancel-button Exit-Script 10 58 2 \
    "0" "KVM64 (Default)" ON \
    "1" "Host" OFF \
    3>&1 1>&2 2>&3); then
    if [ $CPU_TYPE1 = "1" ]; then
      echo -e "${DGN}Using CPU Model: ${BGN}Host${CL}"
      CPU_TYPE=" -cpu host"
    else
      echo -e "${DGN}Using CPU Model: ${BGN}Default${CL}"
      CPU_TYPE=""
    fi
  else
    exit-script
  fi
```

**Reply by tteck** [reply]
In my opinion, selecting `host` as the `CPU_TYPE` would still be the most suitable choice.

**Reply by tteck** [reply]
If you are interested in evaluating any advantages of using `x86-64-v2-AES` as your `CPU_TYPE`, it is readily achievable.

![Screenshot 2023-06-26 4 10 44 PM](https://github.com/tteck/Proxmox/assets/21342670/7377d91c-d041-4717-9381-50d8f2a31497)

**Comment by JPurp** [comment]
Appreciate the quick reply. 

Interestingly in the GUI when creating a new VM it does default to x86-64-v2-AES. Not sure why when initiated from the CLI it defaults to the PVE 7 standard of kvm64.

I do intended to change to the new CPU type at the next reboot and I will report back if I see any issues. 

Thanks

**Reply by tteck** [reply]
I have observed the same thing. Please provide us with your results.

**Reply by tteck** [reply]
In my opinion, I don't see any advantages in using the `x86-64-v2-AES` CPU type over the `host` CPU type in our application.
